src/esp/assets/ResourceManager.cpp

```diff
@@ -1608,7 +1608,7 @@ gfx::PhongMaterialData::uptr ResourceManager::buildFlatShadedMaterialData(
     finalMaterial->ambientTexture =
         textures_.at(textureBaseIndex + material.diffuseTexture()).get();
   } else {
-    finalMaterial->ambientColor = material.ambientColor();
+    finalMaterial->ambientColor = Mn::Color4(1.f, 1.f, 1.f, 1.f);
```

Comment:
May I understand the reason why you would like to change this?

Reply:
I'm not sure the exact reason because @eundersander wrote this part, so maybe he can give an explanation. However, without this change, everything appears black in the Replica dataset mesh example.

Reply:
No, this is the ambientColor (environment color). Do not set it unless the material says so.

Reply:
The ambient color defined in the material is often going to be black, but if we're going to render as flat-shaded, we need ambient color to be set to white. Notice how this function sets ambient color to white by default at the top.

Remember how our "flat-shading" works: we run the phong shader in a special way with no lights, making sure the ambient contribution is appropriate to give the flat-shaded appearance.

We do need this fix for non-textured models to be flat-shaded correctly. We can't leave the code as-is. A cleaner fix would be to simply remove this else block altogether (thus leaving ambient color as white).

Reply:
Note that Lawrence and I have made more progress here. We have a better solution that what I described above, coming soon.